godot-core/src/builtin/arrays.rs

```diff
@@ -19,7 +19,15 @@ use sys::{ffi_methods, interface_fn, GodotFfi};
 ///
 /// Unlike GDScript, all indices and sizes are unsigned, so negative indices are not supported.
 ///
-/// # Safety
+/// # Reference semantics
+///
+/// Like in GDScript, `Array` acts as a reference type: multiple `Array` instances may refer to the
+/// same underlying array, and changes to one are visible in the other.
+///
+/// To create a copy that shares data with the original array, use [`Clone::clone()`]. If you want
+/// to create a copy of the data, use [`duplicate_shallow()`] or [`duplicate_deep()`].
```

Comment:
As mentioned elsewhere, I would not yet implement `Clone` (see below as well).
The `obj::traits::Share` trait might be a candidate for this.

Reply:
`Share` sounds good! Let me do that.